core-features/tracing/index.test.js

```diff
@@ -0,0 +1,447 @@
+      url: `http://localhost:4000/users`,
+    })
+    expect(appResponse.status).toEqual(200)
+
+    const jeagerResponse = await axios({
+      method: 'GET',
+      url: 'http://localhost:16686/api/traces?service=prisma-tracing',
+    })
+    expect(jeagerResponse.status).toEqual(200)
+
+    const [trace] = jeagerResponse.data.data
+    const [rootSpan, ...restOfSpans] = trace.spans
+    const traceTree = buildTree({ span: rootSpan }, restOfSpans)
+
+    const snapshotJSON = JSON.stringify(
+      traceTree,
+      (key, value) => {
+        const ignoredKeys = [
+          'traceID',
+          'spanID',
+          'flags',
+          'references',
+          'startTime',
+          'duration',
+          'warnings',
+          'processID',
+          'logs',
+        ]
+        if (ignoredKeys.includes(key)) {
+          return undefined
+        }
+
+        // If its SQL stmt
+        if (key === 'value' && typeof value === 'string') {
+          if (value.includes('traceparent')) {
+            // Traceparent is unique so we cant snapshot
+            const [sql] = value.split('traceparent')
+            value = sql
+          }
+        }
+
+        return value
+      },
+      2,
+    )
+
+    const prettyJSON = JSON.stringify(JSON.parse(snapshotJSON), null, 2)
+
+    expect(prettyJSON).toMatchInlineSnapshot(`
+      "{
+        \\"span\\": {
+          \\"operationName\\": \\"middleware - runConnect\\",
+          \\"tags\\": [
+            {
+              \\"key\\": \\"fastify.type\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"middleware\\"
+            },
+            {
+              \\"key\\": \\"plugin.name\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"fastify-express\\"
+            },
+            {
+              \\"key\\": \\"hook.name\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"onRequest\\"
+            },
+            {
+              \\"key\\": \\"service.name\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"prisma-tracing\\"
+            },
+            {
+              \\"key\\": \\"telemetry.sdk.language\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"nodejs\\"
+            },
+            {
+              \\"key\\": \\"telemetry.sdk.name\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"opentelemetry\\"
+            },
+            {
+              \\"key\\": \\"telemetry.sdk.version\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"1.4.0\\"
+            },
+            {
+              \\"key\\": \\"otel.library.name\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"@opentelemetry/instrumentation-fastify\\"
+            },
+            {
+              \\"key\\": \\"otel.library.version\\",
+              \\"type\\": \\"string\\",
+              \\"value\\": \\"0.25.0\\"
+            },
```

Comment:
Is `1.4.0` for:
    `"@opentelemetry/exporter-jaeger": "^1.4.0",`

Is `0.25.0` for:
    `"@opentelemetry/instrumentation-fastify": "^0.25.0",`

These versions in the snapshot will be annoying later to maintain as any change will break the test.
Is there another way to test this? or remove the versions / normalize them?

Reply:
Good idea! Removed.